pilot/pkg/model/service.go

```diff
@@ -291,6 +291,16 @@ type NetworkEndpoint struct {
 	// catalog.mystore.com)
 	ServicePort *Port
 
+	// Namespace for the workload on this network endpoint.
+	// For the platforms without namespace, it will be empty.
+	Namespace string
```

Comment:
I think you can combine all the changes to populate these new fields in on PR.

Reply:
Just wondering should these be further wrapped in a `WorkloadAttributes`?

It's the namespace (serviceaccount, labels) of the workload (pod) but without reading the comment, it looks like it's the namespace (serviceaccounts, labels) of the Endpoint itself.

Also note in `ServiceAttributes`, the `Name`, `Namespace` and `UID` are all referring to the service itself.

Reply:
These attributes have been organized into a struct called EndpointAttributes.

Reply:
Added the code to populate the EndpointAttributes in InstancesByPort(). 
Added a test TestServiceEndpointsAttributes() to test the EndpointAttributes of service instances.